packages/mobile/src/screens/favorites-screen/FavoritesScreen.tsx

```diff
@@ -41,13 +53,20 @@ export const FavoritesScreen = () => {
     FeatureFlags.OFFLINE_MODE_ENABLED
   )
 
+  const savedTracks = useProxySelector(getTracks, [])
+
   return (
     <Screen>
       <ScreenHeader
         text={messages.header}
         icon={IconFavorite}
         styles={{ icon: { marginLeft: 3 } }}
-      />
+      >
+        <DownloadToggle
+          collection={DOWNLOAD_REASON_FAVORITES}
+          tracks={savedTracks.entries}
```

Comment:
are all the saved tracks loaded at once on this screen (as opposed to loading in more when you scroll)? also should we disable this if they're not done fetching yet (or the fetch errored out)?

Reply:
No, in fact I think I need to rewrite the downloader to work properly for saves. I'm going to tackle that in a future PR though as I don't currently have the queueing mechanism for downloads. My understanding is that the `favorites` endpoint returns every favorited item (not paginated) so we can queue them, fetch each one from discovery as needed, and save it.